rskj-core/src/main/java/co/rsk/pcc/bto/GetMultisigScriptHash.java

```diff
@@ -0,0 +1,128 @@
+/*
+ * This file is part of RskJ
+ * Copyright (C) 2019 RSK Labs Ltd.
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU Lesser General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+ * GNU Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public License
+ * along with this program. If not, see <http://www.gnu.org/licenses/>.
+ */
+
+package co.rsk.pcc.bto;
+
+import co.rsk.bitcoinj.core.BtcECKey;
+import co.rsk.bitcoinj.script.Script;
+import co.rsk.bitcoinj.script.ScriptBuilder;
+import co.rsk.pcc.ExecutionEnvironment;
+import co.rsk.pcc.NativeContractIllegalArgumentException;
+import co.rsk.pcc.NativeMethod;
+import org.bouncycastle.util.encoders.Hex;
+import org.ethereum.core.CallTransaction;
+
+import java.math.BigInteger;
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.List;
+
+/**
+ * This implements the "getMultisigScriptHash" method
+ * that belongs to the BTOUtils native contract.
+ *
+ * @author Ariel Mendelzon
+ */
+public class GetMultisigScriptHash extends NativeMethod {
+    private final CallTransaction.Function function = CallTransaction.Function.fromSignature(
+            "getMultisigScriptHash",
+            new String[]{"int256", "bytes[]"},
+            new String[]{"bytes"}
+    );
+
+    private final static int COMPRESSED_PUBLIC_KEY_LENGTH = 33;
+    private final static int UNCOMPRESSED_PUBLIC_KEY_LENGTH = 65;
+
+    // Enforced by the 520-byte size limit of the redeem script
+    // (see https://github.com/bitcoin/bips/blob/master/bip-0016.mediawiki#520byte_limitation_on_serialized_script_size)
+    private final static int MAXIMUM_ALLOWED_SIGNATURES = 15;
```

Comment:
bitcoinj's `createMultiSigOutputScript` allows 16 public keys

Reply:
But in the bip0016 they specifically indicate 15 is the maximum. I would prefer to stick to their limit. What do you think?

Reply:
Yes, I commented it just in case 16 is a value used in practice. There is a difference between number of public keys and number of signatures anyway.